### ISSUE TITLE
Pre-configured OIDC server should use OIDC refresh token grant type

### ISSUE THREAD

**Issue by burkel24** [oauthlib/oauthlib#827]
**Describe the feature**

The pre-configured OIDC server class at `openid.connect.core.endpoints.pre_configured` should use the OIDC-specific `RereshTokenGrant` defined in `openid.connect.core.grant_types.refresh_token`.

**Additional context**

#752 Added an OIDC-specific refresh token grant type, which enables ID tokens to be issued as part of the token refresh flow. The pre-configured OIDC server at `oauthlib.openid.connect.core.endpoints.pre_configured`, however, does not take advantage of this OIDC-specific grant type (it just uses the regular rfc6749 `RefreshTokenGrant` instead).

Ultimately, library consumers who wish to issue ID tokens on refresh must create their own custom Server class and a couple properties. As far as I can tell, this process is not documented anywhere and requires digging around in library source code to figure out how to do it.

Intuitively, one would expect the pre-configured OIDC server to use OIDC grant types.

I'm happy to create a PR for this, but could use a bit of guidance re: whether a grant choice dispatcher should be created or not for the refresh token grant type.

**Comment by Pankrat** [comment]
Also stumbled upon this while trying to integrate `oauthlib` via `django-oauth-toolkit` (which uses the pre_configured configuration if OIDC is enabled).

Is there any reason *not* to set `self.refresh_grant` to `RefreshTokenGrant` from `oauthlib.openid.connect.core.grant_types`?

Here's the (somewhat messy) workaround I'm currently using for getting ID tokens when refreshing tokens:

```python
from oauthlib.oauth2.rfc6749.endpoints import TokenEndpoint
from oauthlib.openid.connect.core.endpoints.pre_configured import Server as BaseServer
from oauthlib.openid.connect.core.grant_types import RefreshTokenGrant

class Server(BaseServer):
    def __init__(self, request_validator, *args, **kwargs):
        super().__init__(request_validator, *args, **kwargs)
        self.refresh_grant = RefreshTokenGrant(request_validator)
        TokenEndpoint.__init__(self, default_grant_type='authorization_code',
                               grant_types={
                                   'authorization_code': self.token_grant_choice,
                                   'password': self.password_grant,
                                   'client_credentials': self.credentials_grant,
                                   'refresh_token': self.refresh_grant,
                               },
                               default_token_type=self.bearer)
``` 

I'm not sure if there's an easier way to achieve this. If the openid pre_configured server was changed in oauthlib, all it would take would be to change where `RefreshTokenGrant` is imported from (unless I'm missing something).

**Comment by burkel24** [comment]
@Pankrat the fixed is merged, should be out in the 3.3.0 release

**Comment by hugochinchilla** [comment]
I had to pin this package to the latest revision in master as this fixes a bad behavior of an application for me. It would be very nice if you can release a new version with it, as I see the 3.3.0 milestone has been open for a long time now, maybe the issue holding it back can wait for another release.

Thanks @burkel24 and @nsklikas for enabling refreshing id tokens.
